.github/workflows/pr-preview.yml

```diff
@@ -0,0 +1,99 @@
+name: PR Preview
+
+on:
+  push:
+    branches:
+      - '**'
+
+jobs:
+  pr-preview:
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Create Wheel File
+        timeout-minutes: 120
+        run: |
+          sudo apt install rsync
+          BUILD_AS_FAST_AS_POSSIBLE=1 make package
+      - name: Set PREVIEW_BRANCH & BRANCH env variable
+        uses: ./.github/actions/preview_branch
+        with:
+          pull_request_number: ${{ github.event.pull_request.number }}
+          ref_type: ${{ github.ref_type }}
+          branch: ${{ github.ref_name }}
+
+      # Need the AWS S3 secret/key for the below steps - added the if statement to gate for now
+      - if: ${{ 'true' == 'false' }}
+        name: Upload wheel to S3
+        run: |
+          sudo apt install -y awscli
+
+          aws configure set aws_access_key_id "$CORE_PREVIEWS_S3_KEY_ID"
+          aws configure set aws_secret_access_key "$CORE_PREVIEWS_S3_SECRET_KEY"
```

Comment:
Would probably be caught once actually adding the S3 creds, but this should be accessed via `secrets.*`

Reply:
Oh okay so it is preferable in this case to avoid establishing these as env variables using that syntax for this step? ([lines 73 & 74](https://github.com/streamlit/streamlit/pull/5213/files/1566d8b07dae2d6283f9ec0c58a442b88a418c8d#diff-737d2247cd4b3b6578d2c5413b555a5ca90f6aaa32613770ffdc146900d0469bR73))

Reply:
Whoops, nevermind -- this was just me not quite understanding where/how you're allowed to interpolate values in a GH Actions yaml file.

Looking at the documentation again, it seems like the standard way of making a secret available here would be to either pass it in as an env var (or an input), which is already what's being done here.

In this case, I'd just recommend moving the `env` field above the `run` field (assuming that's legal syntax) to make it a bit more obvious where these values are coming from.

Reply:
That makes sense 👍🏼